File: Backend/app/schemas/department.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class DepartmentBase(BaseModel):
    name: str
    description: Optional[str] = None
    parent_id: Optional[int] = None
    manager_id: Optional[int] = None
    is_active: Optional[bool] = True


class DepartmentCreate(DepartmentBase):
    pass


class DepartmentUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    parent_id: Optional[int] = None
    manager_id: Optional[int] = None
    is_active: Optional[bool] = None


class DepartmentRead(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    parent_id: Optional[int] = None
    manager_id: Optional[int] = None
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class DepartmentTreeNode(DepartmentRead):
    children: List["DepartmentTreeNode"] = []


DepartmentTreeNode.model_rebuild()

File: Frontend/zyro-web/src/app/types/department.ts
export interface Department {
  id: number;
  name: string;
  description?: string | null;
  parent_id?: number | null;
  manager_id?: number | null;
  organization_id: number;
  is_active: boolean;
  created_at: string;
  updated_at?: string | null;
}

export interface DepartmentCreate {
  name: string;
  description?: string;
  parent_id?: number | null;
  manager_id?: number | null;
  organization_id: number;
  is_active?: boolean;
}

export interface DepartmentUpdate {
  name?: string;
  description?: string;
  parent_id?: number | null;
  manager_id?: number | null;
  is_active?: boolean;
}
